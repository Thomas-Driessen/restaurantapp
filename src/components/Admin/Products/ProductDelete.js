import React from "react";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import RemoveIcon from '@material-ui/icons/Remove';
import { IconButton } from "@material-ui/core";

const Product = (props) => {
    const [del, setDelete] = React.useState(false);
    const [backToMenu, setBackToMenu] = React.useState(false);

    const handleDeleteOpen = () => {
        setDelete(true);
    };

    const handleDeleteClose = () => {
        setDelete(false);
    };

    const handleBackToMenuOpen = () => {
        setBackToMenu(true);
    };

    const handleBackToMenuClose = () => {
        setBackToMenu(false);
    };

    const setItemOnMenu = (isOnMenu) => {
        let newProduct = props.product;
        let productType = isOnMenu === 1 ? props.productType : props.product.productType;
        newProduct.onMenu = isOnMenu;
        fetch(`${process.env.REACT_APP_API_URL}/api/${productType}/${newProduct.id}`, {
            method: "PUT",
            mode: "cors",
            headers: {
                Accept: "application/json",
                "Content-Type": "application/json",
            },
            body: JSON.stringify(newProduct),
        }).catch(console.log);

        alert("Your changes have been saved");
        if (isOnMenu === true) {
            setBackToMenu(false);
        }
        else {
            setDelete(false);
        }
        window.location.reload(false);
    };

    return (
        <div>
            {props.product.onMenu ? (
                <IconButton
                    size="medium"
                    color="primary"
                    onClick={handleDeleteOpen}
                >
                    <RemoveIcon />
                </IconButton>
            ) : <Button
                size="large"
                color="primary"
                target="_blank"
                onClick={handleBackToMenuOpen}
            >
                    <span style={{ fontWeight: "bold" }}>Back to menu</span>
                </Button>}

            <Dialog
                open={backToMenu}
                onClose={handleBackToMenuClose}
                aria-labelledby="product-title"
            >
                <DialogTitle id="product-title">
                    {props.product.title}
                </DialogTitle>
                <DialogContent>
                    <DialogContentText>
                        Are you sure you want to bring this product back to the menu?
                      </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button
                        size="medium"
                        color="primary"
                        onClick={() => setItemOnMenu(true)}
                    >
                        Yes
                      </Button>
                    <Button
                        size="medium"
                        color="primary"
                        onClick={handleBackToMenuClose}
                    >
                        No
                      </Button>
                </DialogActions>
            </Dialog>

            <Dialog
                open={del}
                onClose={handleDeleteClose}
                aria-labelledby="product-title"
            >
                <DialogTitle id="product-title">
                    {props.product.title}
                </DialogTitle>
                <DialogContent>
                    <DialogContentText>
                        Are you sure you want to delete this product?
                      </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button
                        size="medium"
                        color="primary"
                        onClick={() => setItemOnMenu(false)}
                    >
                        Yes
                      </Button>
                    <Button
                        size="medium"
                        color="primary"
                        onClick={handleDeleteClose}
                    >
                        No
                      </Button>
                </DialogActions>
            </Dialog>
        </div>
    );
};
export default Product;
