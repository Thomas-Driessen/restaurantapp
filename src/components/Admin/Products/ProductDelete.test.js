import React from 'react';
import { render } from '@testing-library/react';
import ProductDelete from './ProductDelete';
import renderer from 'react-test-renderer';

const product = {
    id: 1,
    title: "Coca-Cola",
    price: 2.5,
    description: "",
    category: { categoryName: "Drinks" },
    onMenu: true
};

const productNotOnMenu = {
    id: 1,
    title: "Coca-Cola",
    price: 2.5,
    description: "",
    category: { categoryName: "Drinks" },
    onMenu: false
};

it('ProductDelete renders correctly', () => {
    const tree = renderer
        .create(<ProductDelete product={product} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('Render ProductDelete NotOnMenu', () => {
    const { getByText } = render(<ProductDelete product={productNotOnMenu} />);
    const linkElement = getByText(/Back to menu/i);
    expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
    global.gc && global.gc()
})