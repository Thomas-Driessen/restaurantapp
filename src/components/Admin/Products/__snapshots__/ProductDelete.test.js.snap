// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDelete renders correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13H5v-2h14v2z"
        />
      </svg>
    </span>
  </button>
</div>
`;
